hw_4_js/exercises.js

```diff
@@ -0,0 +1,415 @@
+/**
+ * Exercise 1
+ *
+ * Write a function that returns odd array values.
+ * [1,2,3,4] => [1,3]
+ */
+
+const getOddValues = (numbers) => {
+  let odds = []
+  for (const number of numbers) {
+    if (number % 2 != 0) {
+      odds.push(number)
+    }
+  }
+  return odds
+}
+
+/**
+ * Exercise 2
+ *
+ * Write a function that returns the smallest value of an array
+ * [4,2,10,27] => 2
+ */
+
+const getSmallestValue = (numbers) => {
+  let min = numbers[0]
+  for (let i = 1; i < numbers.length; i++) {
+    if (numbers[i] < min) {
+      min = numbers[i]
+    }
+  }
+  return min
+}
+
+/**
+ * Exercise 3
+ *
+ * Write a function that returns the biggest value of an array
+ * [5,22,9,43] => 43
+ */
+
+const getBiggestValue = (numbers) => {
+  let max = numbers[0]
+  for (let i = 1; i < numbers.length; i++) {
+    if (numbers[i] > max) {
+      max = numbers[i]
+    }
+  }
+  return max
+}
+
+/**
+ * Exercise 4
+ *
+ * Write a function that takes an array of strings as input
+ * and returns only those shorter than 20 characters
+ *
+ *[
+ * 'I am a short string',
+ * 'I seem to be short too',
+ * 'And I am a long string'
+ *] => [
+ * 'I am a short string',
+ * 'And I am a long string'
+ *]
+ *
+ * Use: filter
+ */
+const getShorterStrings = (strings, characters = 20) => {
+  return strings.filter(string => string.length < characters)
+}
+
+/**
+ * Exercise 5
+ *
+ * Write a function that takes the following data as input:
+ *
+ *[
+ * { name: 'shark', likes: 'ocean' },
+ * { name: 'turtle', likes: 'pond' },
+ * { name: 'otter', likes: 'fish biscuits' }
+ *]
+ *
+ * And returns an array of strings:
+ *
+ * [ 'shark likes ocean', 'turtle likes pond', 'otter likes fish biscuits' ]
+ *
+ * Use: map
+ */
+
+const getComputedStrings = (fish) => {
+  return fish.map(({ name, likes }) => name + ' likes ' + likes)
```

Comment:
```suggestion
  return fish.map(({ name, likes }) => `${name} likes ${likes}`)
```
It is the preferable way in modern JS